### PR TITLE
MetadataReader: Allow for an unresolved pointer to be returned by getSymbol

### PR DESCRIPTION
**Pull request by aschwaighofer**
rdar://122506294

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci test

File: include/swift/Remote/MetadataReader.h [hunk 1]
```diff
@@ -517,13 +517,30 @@ class MetadataReader {
         // 'resolved' points to a struct of two relative addresses.
         // The second entry is a relative address to the mangled protocol
         // without symbolic references.
+
+        // lldb might return and unresolved remote absolute pointer from its
```
Comment (adrian-prantl) [inline include/swift/Remote/MetadataReader.h:521]:
```suggestion
        // lldb might return an unresolved remote absolute pointer from its
```

**Review by kastiglione — APPROVED** [review]

**Comment by aschwaighofer** [comment]
@swift-ci test

**Comment by aschwaighofer** [comment]
@swift-ci test linux

**Comment by aschwaighofer** [comment]
@swift-ci test linux

**Comment by kastiglione** [comment]
@swift-ci test linux

**Comment by aschwaighofer** [comment]
@swift-ci test linux

**Pull request merged**